### PR TITLE
dev(feat): Various improvements to script

### PR DESCRIPTION
**Pull request by armenzg**
Included as part of the Apple arm64 work since these changes were part of re-validating the bootstrap script.

### CODE REVIEW

File: bootstrap.sh [hunk 1]
```diff
@@ -298,7 +298,7 @@ install_homebrew() {
   # Update Homebrew.
   export PATH="$HOMEBREW_PREFIX/bin:$PATH"
   logn "Updating Homebrew:"
-  brew update --quiet
+  [ -z "$QUICK" ] && brew update --quiet
```
Comment (armenzg) [inline bootstrap.sh:301]: This is useful when I want faster runs (e.g. `QUICK=on ./bootstraph.sh`

File: bootstrap.sh [hunk 2]
```diff
@@ -429,12 +429,21 @@ ensure_docker_server() {
...
     log "Installing from sentry Brewfile (very slow)"
-    # This is useful when trying to run the script on a non-clean machine
-    (cd "$1" && brew bundle) || log "Something failed during brew bundle but let's try to continue"
+    if [ -z "$QUICK" ]; then
+      # This makes sure that we don't try to install Python packages from the cache
+      # This is useful when verifying a new platform and multiple executions of bootstrap.sh is required
+      export PIP_NO_CACHE_DIR=on
+      # The fallback is useful when trying to run the script on a non-clean machine multiple times
+      cd "$1" && (make prerequisites || log "Something failed during brew bundle but let's try to continue")
+    else
```
Comment (armenzg) [inline bootstrap.sh:442]: This block is useful when running `./uninstall.sh && QUICK=on ./bootstrap.sh`.

File: bootstrap.sh [hunk 3]
```diff
@@ -540,6 +549,8 @@ bootstrap() {
 ## Beginning of execution ##
 ############################
 
+install_sentry_cli
```
Comment (armenzg) [inline bootstrap.sh:552]: Moving this higher up since we had been loosing observability.

File: bootstrap.sh [hunk 4]
```diff
@@ -602,13 +612,13 @@ export PYENV_VERSION=$(
   get-pyenv-version
 )
 setup_virtualenv "$SENTRY_ROOT"
-# Currently failling in CI; fix at later time
-[ -z "$CI" ] && install_volta
```
Comment (armenzg) [inline bootstrap.sh:606]: Moving it below after the point where we exit the CI.

File: uninstall.sh [hunk 5]
```diff
@@ -1,18 +1,29 @@
...
-[ -d /Applications/Docker.app ] && rm -rf /Applications/Docker.app
+# Remove Sentry and dependencies
+rm -rf ~/.sentry
+rm -rf ~/code/sentry/{.venv,node_modules}
+rm -rf ~/code/getsentry/{.venv,node_modules}
+[ -f ~/.zprofile ] && rm ~/.zprofile
 
-# Since we install pre-commit via brew, we need to restore the hooks
-[ -f .git/hooks/pre-commit ] && pre-commit uninstall
+remove_brew_setup() {
```
Comment (armenzg) [inline uninstall.sh:14]: This does a more thorough clean-up than just calling the official uninstall script.

File: bootstrap.sh [hunk 6]
```diff
@@ -429,12 +429,21 @@ ensure_docker_server() {
...
 }
 
-# Install Brewfile of dir ($1)
-install_brewfile() {
+# Install required libraries of dir ($1)
+install_prerequisites() {
   if [ -d "$1" ]; then
     log "Installing from sentry Brewfile (very slow)"
-    # This is useful when trying to run the script on a non-clean machine
-    (cd "$1" && brew bundle) || log "Something failed during brew bundle but let's try to continue"
```
Comment (billyvg) [inline bootstrap.sh:437]: Do we no longer install deps from Brewfile?

File: uninstall.sh [hunk 7]
```diff
@@ -1,18 +1,29 @@
...
+# This script tries to remove as much as reasonable to allow for re-testing
+# bootstrap.sh multiple times on the same host
 
-# brew's uninstall script does not properly remove casks
-command -v brew &>/dev/null && brew uninstall --cask docker
-[ -d /Applications/Docker.app ] && rm -rf /Applications/Docker.app
+# Remove Sentry and dependencies
+rm -rf ~/.sentry
+rm -rf ~/code/sentry/{.venv,node_modules}
+rm -rf ~/code/getsentry/{.venv,node_modules}
```
Comment (billyvg) [inline uninstall.sh:11]: Should these paths be hard coded?

File: uninstall.sh [hunk 8]
```diff
@@ -1,18 +1,29 @@
...
+#          this script. This script has only been tested on Armen's MBP arm64 machine
+#
+# This script tries to remove as much as reasonable to allow for re-testing
+# bootstrap.sh multiple times on the same host
 
-# brew's uninstall script does not properly remove casks
-command -v brew &>/dev/null && brew uninstall --cask docker
-[ -d /Applications/Docker.app ] && rm -rf /Applications/Docker.app
+# Remove Sentry and dependencies
+rm -rf ~/.sentry
```
Comment (billyvg) [inline uninstall.sh:9]: I'd prefer not removing this, for example, I believe the sentry configs are stored here and I wouldn't want to inadvertently remove them.

Comment (billyvg) [reply uninstall.sh:9]: Oh I guess this isn't meant to be public and only used for testing?

**Pull request merged**